[Tellstick] NPE's after OH upgrade

After upgrading my OH environment from snapshot 1103 to 1114, Tellstick binding doesn't seem to work anymore. My Tellstick Duo Thing never leaves the INITIALIZING state.

When restarting the binding I get the following in my log:

```
20:00:42.587 [INFO ] [ome.event.ThingStatusInfoChangedEvent] - 'tellstick:telldus-core:1' changed from INITIALIZING to UNINITIALIZED
20:00:42.591 [ERROR] [rnal.common.AbstractInvocationHandler] - An error occurred while calling method 'ThingHandler.dispose()' on 'org.openhab.binding.tellstick.internal.core.TelldusCoreBridgeHandler@6032bb': null
java.lang.NullPointerException: null
        at org.openhab.binding.tellstick.internal.core.TelldusCoreBridgeHandler.dispose(TelldusCoreBridgeHandler.java:81) [213:org.openhab.binding.tellstick:2.2.0.201712052342]
        at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method) ~[?:?]
        at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62) ~[?:?]
        at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43) ~[?:?]
        at java.lang.reflect.Method.invoke(Method.java:498) ~[?:?]
        at org.eclipse.smarthome.core.internal.common.AbstractInvocationHandler.invokeDirect(AbstractInvocationHandler.java:157) [112:org.eclipse.smarthome.core:0.9.0.201712011551]
        at org.eclipse.smarthome.core.internal.common.TrackingCallable.call(TrackingCallable.java:45) [112:org.eclipse.smarthome.core:0.9.0.201712011551]
        at java.util.concurrent.FutureTask.run(FutureTask.java:266) [?:?]
        at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149) [?:?]
        at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624) [?:?]
        at java.lang.Thread.run(Thread.java:748) [?:?]
20:00:42.716 [ERROR] [home.core.thing.internal.ThingManager] - Exception occurred while disposing handler of thing 'tellstick:telldus-core:1': null
java.lang.NullPointerException: null
        at org.openhab.binding.tellstick.internal.core.TelldusCoreBridgeHandler.dispose(TelldusCoreBridgeHandler.java:81) [213:org.openhab.binding.tellstick:2.2.0.201712052342]
        at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method) ~[?:?]
        at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62) ~[?:?]
        at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43) ~[?:?]
        at java.lang.reflect.Method.invoke(Method.java:498) ~[?:?]
        at org.eclipse.smarthome.core.internal.common.AbstractInvocationHandler.invokeDirect(AbstractInvocationHandler.java:157) [112:org.eclipse.smarthome.core:0.9.0.201712011551]
        at org.eclipse.smarthome.core.internal.common.TrackingCallable.call(TrackingCallable.java:45) [112:org.eclipse.smarthome.core:0.9.0.201712011551]
        at java.util.concurrent.FutureTask.run(FutureTask.java:266) [?:?]
        at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149) [?:?]
        at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624) [?:?]
        at java.lang.Thread.run(Thread.java:748) [?:?]
20:00:42.844 [INFO ] [ome.event.ThingStatusInfoChangedEvent] - 'tellstick:telldus-core:1' changed from UNINITIALIZED to UNINITIALIZED (HANDLER_MISSING_ERROR)
20:00:43.358 [INFO ] [ome.event.ThingStatusInfoChangedEvent] - 'tellstick:telldus-core:1' changed from UNINITIALIZED (HANDLER_MISSING_ERROR) to INITIALIZING
20:00:48.361 [WARN ] [rnal.common.AbstractInvocationHandler] - Timeout of 5000ms exceeded while calling method 'ThingHandler.initialize()' on 'org.openhab.binding.tellstick.internal.core.TelldusCoreBridgeHandler@5af8af'. Thread 'safeCall-27' (701) is in state 'RUNNABLE'
        at java.lang.Class.isArray(Native Method)
        at com.sun.jna.Function.isPrimitiveArray(Function.java:625)
        at com.sun.jna.Function.convertArgument(Function.java:498)
        at com.sun.jna.Function.invoke(Function.java:297)
        at com.sun.jna.Library$Handler.invoke(Library.java:212)
        at com.sun.proxy.$Proxy148.tdSensor(Unknown Source)
        at org.tellstick.device.TellstickSensor.getAllSensors(TellstickSensor.java:48)
        at org.openhab.binding.tellstick.internal.core.TelldusCoreBridgeHandler.rescanTelldusDevices(TelldusCoreBridgeHandler.java:136)
        at org.openhab.binding.tellstick.internal.core.TelldusCoreBridgeHandler.initialize(TelldusCoreBridgeHandler.java:111)
        at sun.reflect.GeneratedMethodAccessor56.invoke(Unknown Source)
        at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
        at java.lang.reflect.Method.invoke(Method.java:498)
        at org.eclipse.smarthome.core.internal.common.AbstractInvocationHandler.invokeDirect(AbstractInvocationHandler.java:157)
        at org.eclipse.smarthome.core.internal.common.TrackingCallable.call(TrackingCallable.java:45)
        at java.util.concurrent.FutureTask.run(FutureTask.java:266)
        ...
20:00:48.650 [WARN ] [home.core.thing.internal.ThingManager] - Initializing handler for thing 'tellstick:telldus-core:1' takes more than 5000ms.
```
(Wow, that's hard to read in the preview. Any better way to put log snippets in issues?)

Also tried with debug logging on, got nothing more useful there.

I noticed a setting on the gateway Thing for library file location. Don't know if that one is new, never seen it before. It was set to C:\ something which is clearly wrong since I'm on Linux. Tried various ways of pointing out the libtelldus-core.so though, didn't make any difference. Might be completely unrelated...

This is (ARM) machine running Debian Jessie. OH installed from repository.

```
22:19:44.248 [ERROR] [al.discovery.TellstickBridgeDiscovery] - Could not load telldus core, please make sure Telldus is installed and correct 32/64 bit java.
java.lang.NoClassDefFoundError: Could not initialize class org.tellstick.JNA$CLibrary
        at org.tellstick.device.TellstickController.getControllers(TellstickController.java:56) ~[202:org.openhab.binding.tellstick:2.2.0.201712041557]
        at org.openhab.binding.tellstick.internal.discovery.TellstickBridgeDiscovery.listBridge(TellstickBridgeDiscovery.java:73) [202:org.openhab.binding.tellstick:2.2.0.201712041557]
        at org.openhab.binding.tellstick.internal.discovery.TellstickBridgeDiscovery.discoverController(TellstickBridgeDiscovery.java:66) [202:org.openhab.binding.tellstick:2.2.0.201712041557]
        at org.openhab.binding.tellstick.internal.discovery.TellstickBridgeDiscovery.startScan(TellstickBridgeDiscovery.java:50) [202:org.openhab.binding.tellstick:2.2.0.201712041557]
        at org.eclipse.smarthome.config.discovery.AbstractDiscoveryService.startScan(AbstractDiscoveryService.java:222) [104:org.eclipse.smarthome.config.discovery:0.9.0.201712011551]
        at org.eclipse.smarthome.config.discovery.internal.DiscoveryServiceRegistryImpl.startScan(DiscoveryServiceRegistryImpl.java:409) [104:org.eclipse.smarthome.config.discovery:0.9.0.201712011551]
        at org.eclipse.smarthome.config.discovery.internal.DiscoveryServiceRegistryImpl.startScans(DiscoveryServiceRegistryImpl.java:385) [104:org.eclipse.smarthome.config.discovery:0.9.0.201712011551]
        at org.eclipse.smarthome.config.discovery.internal.DiscoveryServiceRegistryImpl.startScan(DiscoveryServiceRegistryImpl.java:241) [104:org.eclipse.smarthome.config.discovery:0.9.0.201712011551]
        at org.eclipse.smarthome.io.rest.core.internal.discovery.DiscoveryResource.scan(DiscoveryResource.java:97) [126:org.eclipse.smarthome.io.rest.core:0.9.0.201712011551]
        at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method) ~[?:?]
        at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62) ~[?:?]
        at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43) ~[?:?]
        at java.lang.reflect.Method.invoke(Method.java:497) ~[?:?]
        at org.glassfish.jersey.server.model.internal.ResourceMethodInvocationHandlerFactory$1.invoke(ResourceMethodInvocationHandlerFactory.java:81) [178:org.glassfish.jersey.core.jersey-server:2.22.2]
        at org.glassfish.jersey.server.model.internal.AbstractJavaResourceMethodDispatcher$1.run(AbstractJavaResourceMethodDispatcher.java:144) [178:org.glassfish.jersey.core.jersey-server:2.22.2]
        at org.glassfish.jersey.server.model.internal.AbstractJavaResourceMethodDispatcher.invoke(AbstractJavaResourceMethodDispatcher.java:161) [178:org.glassfish.jersey.core.jersey-server:2.22.2]
        at org.glassfish.jersey.server.model.internal.JavaResourceMethodDispatcherProvider$ResponseOutInvoker.doDispatch(JavaResourceMethodDispatcherProvider.java:160) [178:org.glassfish.jersey.core.jersey-server:2.22.2]
        at org.glassfish.jersey.server.model.internal.AbstractJavaResourceMethodDispatcher.dispatch(AbstractJavaResourceMethodDispatcher.java:99) [178:org.glassfish.jersey.core.jersey-server:2.22.2]
        at org.glassfish.jersey.server.model.ResourceMethodInvoker.invoke(ResourceMethodInvoker.java:389) [178:org.glassfish.jersey.core.jersey-server:2.22.2]
        at org.glassfish.jersey.server.model.ResourceMethodInvoker.apply(ResourceMethodInvoker.java:347) [178:org.glassfish.jersey.core.jersey-server:2.22.2]
        at org.glassfish.jersey.server.model.ResourceMethodInvoker.apply(ResourceMethodInvoker.java:102) [178:org.glassfish.jersey.core.jersey-server:2.22.2]
        at org.glassfish.jersey.server.ServerRuntime$2.run(ServerRuntime.java:326) [178:org.glassfish.jersey.core.jersey-server:2.22.2]
        at org.glassfish.jersey.internal.Errors$1.call(Errors.java:271) [177:org.glassfish.jersey.core.jersey-common:2.22.2]
        at org.glassfish.jersey.internal.Errors$1.call(Errors.java:267) [177:org.glassfish.jersey.core.jersey-common:2.22.2]
        at org.glassfish.jersey.internal.Errors.process(Errors.java:315) [177:org.glassfish.jersey.core.jersey-common:2.22.2]
        at org.glassfish.jersey.internal.Errors.process(Errors.java:297) [177:org.glassfish.jersey.core.jersey-common:2.22.2]
        at org.glassfish.jersey.internal.Errors.process(Errors.java:267) [177:org.glassfish.jersey.core.jersey-common:2.22.2]
        at org.glassfish.jersey.process.internal.RequestScope.runInScope(RequestScope.java:317) [177:org.glassfish.jersey.core.jersey-common:2.22.2]
        at org.glassfish.jersey.server.ServerRuntime.process(ServerRuntime.java:305) [178:org.glassfish.jersey.core.jersey-server:2.22.2]
        at org.glassfish.jersey.server.ApplicationHandler.handle(ApplicationHandler.java:1154) [178:org.glassfish.jersey.core.jersey-server:2.22.2]
        at org.glassfish.jersey.servlet.WebComponent.serviceImpl(WebComponent.java:473) [175:org.glassfish.jersey.containers.jersey-container-servlet-core:2.22.2]
        at org.glassfish.jersey.servlet.WebComponent.service(WebComponent.java:427) [175:org.glassfish.jersey.containers.jersey-container-servlet-core:2.22.2]
        at org.glassfish.jersey.servlet.ServletContainer.service(ServletContainer.java:388) [175:org.glassfish.jersey.containers.jersey-container-servlet-core:2.22.2]
        at org.glassfish.jersey.servlet.ServletContainer.service(ServletContainer.java:341) [175:org.glassfish.jersey.containers.jersey-container-servlet-core:2.22.2]
        at org.glassfish.jersey.servlet.ServletContainer.service(ServletContainer.java:228) [175:org.glassfish.jersey.containers.jersey-container-servlet-core:2.22.2]
        at com.eclipsesource.jaxrs.publisher.internal.ServletContainerBridge.service(ServletContainerBridge.java:76) [15:com.eclipsesource.jaxrs.publisher:5.3.1.201602281253]
        at org.eclipse.jetty.servlet.ServletHolder.handle(ServletHolder.java:848) [88:org.eclipse.jetty.servlet:9.3.22.v20171030]
        at org.eclipse.jetty.servlet.ServletHandler.doHandle(ServletHandler.java:584) [88:org.eclipse.jetty.servlet:9.3.22.v20171030]
        at org.ops4j.pax.web.service.jetty.internal.HttpServiceServletHandler.doHandle(HttpServiceServletHandler.java:71) [191:org.ops4j.pax.web.pax-web-jetty:6.0.7]
        at org.eclipse.jetty.server.handler.ScopedHandler.handle(ScopedHandler.java:143) [87:org.eclipse.jetty.server:9.3.22.v20171030]
        at org.eclipse.jetty.security.SecurityHandler.handle(SecurityHandler.java:548) [85:org.eclipse.jetty.security:9.3.22.v20171030]
        at org.eclipse.jetty.server.session.SessionHandler.doHandle(SessionHandler.java:226) [87:org.eclipse.jetty.server:9.3.22.v20171030]
        at org.eclipse.jetty.server.handler.ContextHandler.doHandle(ContextHandler.java:1180) [87:org.eclipse.jetty.server:9.3.22.v20171030]
        at org.ops4j.pax.web.service.jetty.internal.HttpServiceContext.doHandle(HttpServiceContext.java:284) [191:org.ops4j.pax.web.pax-web-jetty:6.0.7]
        at org.eclipse.jetty.servlet.ServletHandler.doScope(ServletHandler.java:512) [88:org.eclipse.jetty.servlet:9.3.22.v20171030]
        at org.eclipse.jetty.server.session.SessionHandler.doScope(SessionHandler.java:185) [87:org.eclipse.jetty.server:9.3.22.v20171030]
        at org.eclipse.jetty.server.handler.ContextHandler.doScope(ContextHandler.java:1112) [87:org.eclipse.jetty.server:9.3.22.v20171030]
        at org.eclipse.jetty.server.handler.ScopedHandler.handle(ScopedHandler.java:141) [87:org.eclipse.jetty.server:9.3.22.v20171030]
        at org.ops4j.pax.web.service.jetty.internal.JettyServerHandlerCollection.handle(JettyServerHandlerCollection.java:80) [191:org.ops4j.pax.web.pax-web-jetty:6.0.7]
        at org.eclipse.jetty.server.handler.HandlerWrapper.handle(HandlerWrapper.java:134) [87:org.eclipse.jetty.server:9.3.22.v20171030]
        at org.eclipse.jetty.server.Server.handle(Server.java:534) [87:org.eclipse.jetty.server:9.3.22.v20171030]
        at org.eclipse.jetty.server.HttpChannel.handle(HttpChannel.java:333) [87:org.eclipse.jetty.server:9.3.22.v20171030]
        at org.eclipse.jetty.server.HttpConnection.onFillable(HttpConnection.java:251) [87:org.eclipse.jetty.server:9.3.22.v20171030]
        at org.eclipse.jetty.io.AbstractConnection$ReadCallback.succeeded(AbstractConnection.java:283) [79:org.eclipse.jetty.io:9.3.22.v20171030]
        at org.eclipse.jetty.io.FillInterest.fillable(FillInterest.java:108) [79:org.eclipse.jetty.io:9.3.22.v20171030]
        at org.eclipse.jetty.io.SelectChannelEndPoint$2.run(SelectChannelEndPoint.java:93) [79:org.eclipse.jetty.io:9.3.22.v20171030]
        at org.eclipse.jetty.util.thread.strategy.ExecuteProduceConsume.executeProduceConsume(ExecuteProduceConsume.java:303) [90:org.eclipse.jetty.util:9.3.22.v20171030]
        at org.eclipse.jetty.util.thread.strategy.ExecuteProduceConsume.produceConsume(ExecuteProduceConsume.java:148) [90:org.eclipse.jetty.util:9.3.22.v20171030]
        at org.eclipse.jetty.util.thread.strategy.ExecuteProduceConsume.run(ExecuteProduceConsume.java:136) [90:org.eclipse.jetty.util:9.3.22.v20171030]
        at org.eclipse.jetty.util.thread.QueuedThreadPool.runJob(QueuedThreadPool.java:671) [90:org.eclipse.jetty.util:9.3.22.v20171030]
        at org.eclipse.jetty.util.thread.QueuedThreadPool$2.run(QueuedThreadPool.java:589) [90:org.eclipse.jetty.util:9.3.22.v20171030]
        at java.lang.Thread.run(Thread.java:745) [?:?]
```

@kaikreuzer I have tested the changes above and it looks OK, or at least not worse than before :) Only tested with Telldus Net. 

Bridge tellstick:telldus-live:2 "Tellstick Net" [refreshInterval=30000 ...] 

Some notes on the binding behaviour:

1. At start, the following error occurs. I guess its due to the fact that the binding tries to load what's needed for running a Tellstick duo (telldus-core) even though I have configured it to use Tellstick net. (telldus-live)  

```
2017-12-11 22:04:05.771 [ERROR] [t.i.d.TellstickBridgeDiscovery:78   ] - Could not load telldus core, please make sure Telldus is installed and correct 32/64 bit java. java.lang.UnsatisfiedLinkError: Unable to load library 'TelldusCore': Native library (win32-x86-64/TelldusCore.dll) not found in resource path (C:\Users\Jakob\openhab2-master\ws\.metadata\.plugins\org.eclipse.pde.core\.bundle_pool\plugins\org.eclipse.equinox.launcher_1.4.0.v20161219-1356.jar)
	at com.sun.jna.NativeLibrary.loadLibrary(NativeLibrary.java:271)
	at com.sun.jna.NativeLibrary.getInstance(NativeLibrary.java:398)
	at com.sun.jna.Library$Handler.<init>(Library.java:147)
	at com.sun.jna.Native.loadLibrary(Native.java:412)
	at com.sun.jna.Native.loadLibrary(Native.java:391)
	at org.tellstick.JNA$CLibrary.<clinit>(JNA.java:53)
	at org.tellstick.device.TellstickController.getControllers(TellstickController.java:56)
	at org.openhab.binding.tellstick.internal.discovery.TellstickBridgeDiscovery.listBridge(TellstickBridgeDiscovery.java:73)
	at org.openhab.binding.tellstick.internal.discovery.TellstickBridgeDiscovery.discoverController(TellstickBridgeDiscovery.java:66)
	at org.openhab.binding.tellstick.internal.discovery.TellstickBridgeDiscovery.startBackgroundDiscovery(TellstickBridgeDiscovery.java:55)
	at org.eclipse.smarthome.config.discovery.AbstractDiscoveryService.activate(AbstractDiscoveryService.java:385) 
```

2. There is a warning shown multiple times. Two occurrences below. I have 22 devices in my setup. Don't know why this happens...

```
2017-12-11 22:04:20.850 [WARN ] [.i.c.AbstractInvocationHandler:113  ] - Timeout of 5000ms exceeded while calling method 'ThingHandler.initialize()' on 'org.openhab.binding.tellstick.handler.TelldusDevicesHandler@229e749d'. Thread 'safeCall-1' (82) is in state 'WAITING'
	at sun.misc.Unsafe.park(Native Method)
	at java.util.concurrent.locks.LockSupport.park(LockSupport.java:175)
	at java.util.concurrent.locks.AbstractQueuedSynchronizer.parkAndCheckInterrupt(AbstractQueuedSynchronizer.java:836)
	at java.util.concurrent.locks.AbstractQueuedSynchronizer.acquireQueued(AbstractQueuedSynchronizer.java:870)
	at java.util.concurrent.locks.AbstractQueuedSynchronizer.acquire(AbstractQueuedSynchronizer.java:1199)
	at java.util.concurrent.locks.ReentrantLock$NonfairSync.lock(ReentrantLock.java:209)
	at java.util.concurrent.locks.ReentrantLock.lock(ReentrantLock.java:285)
	at org.eclipse.smarthome.core.thing.internal.ThingManager.thingUpdated(ThingManager.java:406)
	at org.eclipse.smarthome.core.thing.internal.ThingRegistryImpl.notifyTrackers(ThingRegistryImpl.java:222)
	at org.eclipse.smarthome.core.thing.internal.ThingRegistryImpl.notifyListenersAboutUpdatedElement(ThingRegistryImpl.java:145)
	at org.eclipse.smarthome.core.thing.internal.ThingRegistryImpl.notifyListenersAboutUpdatedElement(ThingRegistryImpl.java:1)
	at org.eclipse.smarthome.core.common.registry.AbstractRegistry.updated(AbstractRegistry.java:188)
	at org.eclipse.smarthome.core.common.registry.AbstractRegistry.updated(AbstractRegistry.java:1)
	at org.eclipse.smarthome.core.common.registry.AbstractProvider.notifyListeners(AbstractProvider.java:62)
	at org.eclipse.smarthome.core.common.registry.AbstractProvider.notifyListenersAboutUpdatedElement(AbstractProvider.java:86)
	at org.eclipse.smarthome.core.common.registry.AbstractManagedProvider.update(AbstractManagedProvider.java:132)
	at org.eclipse.smarthome.core.thing.internal.ThingManager$1$1.run(ThingManager.java:233)
	at org.eclipse.smarthome.core.thing.internal.ThingManager$1$1.run(ThingManager.java:1)
	at java.security.AccessController.doPrivileged(Native Method)
	...
2017-12-11 22:04:20.850 [WARN ] [.i.c.AbstractInvocationHandler:113  ] - Timeout of 5000ms exceeded while calling method 'ThingHandler.initialize()' on 'org.openhab.binding.tellstick.handler.TelldusDevicesHandler@d8d08b5'. Thread 'safeCall-3' (86) is in state 'WAITING'
	at sun.misc.Unsafe.park(Native Method)
	at java.util.concurrent.locks.LockSupport.park(LockSupport.java:175)
	at java.util.concurrent.locks.AbstractQueuedSynchronizer.parkAndCheckInterrupt(AbstractQueuedSynchronizer.java:836)
	at java.util.concurrent.locks.AbstractQueuedSynchronizer.acquireQueued(AbstractQueuedSynchronizer.java:870)
	at java.util.concurrent.locks.AbstractQueuedSynchronizer.acquire(AbstractQueuedSynchronizer.java:1199)
	at java.util.concurrent.locks.ReentrantLock$NonfairSync.lock(ReentrantLock.java:209)
	at java.util.concurrent.locks.ReentrantLock.lock(ReentrantLock.java:285)
	at org.eclipse.smarthome.core.thing.internal.ThingManager.thingUpdated(ThingManager.java:406)
	at org.eclipse.smarthome.core.thing.internal.ThingRegistryImpl.notifyTrackers(ThingRegistryImpl.java:222)
	at org.eclipse.smarthome.core.thing.internal.ThingRegistryImpl.notifyListenersAboutUpdatedElement(ThingRegistryImpl.java:145)
	at org.eclipse.smarthome.core.thing.internal.ThingRegistryImpl.notifyListenersAboutUpdatedElement(ThingRegistryImpl.java:1)
	at org.eclipse.smarthome.core.common.registry.AbstractRegistry.updated(AbstractRegistry.java:188)
	at org.eclipse.smarthome.core.common.registry.AbstractRegistry.updated(AbstractRegistry.java:1)
	at org.eclipse.smarthome.core.common.registry.AbstractProvider.notifyListeners(AbstractProvider.java:62)
	at org.eclipse.smarthome.core.common.registry.AbstractProvider.notifyListenersAboutUpdatedElement(AbstractProvider.java:86)
	at org.eclipse.smarthome.core.common.registry.AbstractManagedProvider.update(AbstractManagedProvider.java:132)
	at org.eclipse.smarthome.core.thing.internal.ThingManager$1$1.run(ThingManager.java:233)
	at org.eclipse.smarthome.core.thing.internal.ThingManager$1$1.run(ThingManager.java:1)
	at java.security.AccessController.doPrivileged(Native Method)
	...
2017-12-11 22:04:20.850 [WARN ] [.c.thing.internal.ThingManager:622  ] - Initializing handler for thing 'tellstick:switch:2:414128' takes more than 5000ms.
2017-12-11 22:04:20.851 [WARN ] [.c.thing.internal.ThingManager:622  ] - Initializing handler for thing 'tellstick:switch:2:1930308' takes more than 5000ms.
```

3. Got yet another warning...

```
2017-12-11 22:07:24.734 [WARN ] [.l.TelldusLiveDeviceController:285  ] - TimeoutException error in getjava.util.concurrent.TimeoutException: null
	at org.asynchttpclient.netty.NettyResponseFuture.get(NettyResponseFuture.java:169)
	at org.openhab.binding.tellstick.internal.live.TelldusLiveDeviceController.innerCallRest(TelldusLiveDeviceController.java:308)
	at org.openhab.binding.tellstick.internal.live.TelldusLiveDeviceController.callRestMethod(TelldusLiveDeviceController.java:282)
	at org.openhab.binding.tellstick.internal.live.TelldusLiveBridgeHandler.updateDevices(TelldusLiveBridgeHandler.java:132)
	at org.openhab.binding.tellstick.internal.live.TelldusLiveBridgeHandler.refreshDeviceList(TelldusLiveBridgeHandler.java:119)
	at org.openhab.binding.tellstick.internal.live.TelldusLiveBridgeHandler$1.run(TelldusLiveBridgeHandler.java:70)
	at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:511)
	at java.util.concurrent.FutureTask.runAndReset(FutureTask.java:308)
	at java.util.concurrent.ScheduledThreadPoolExecutor$ScheduledFutureTask.access$301(ScheduledThreadPoolExecutor.java:180)
	at java.util.concurrent.ScheduledThreadPoolExecutor$ScheduledFutureTask.run(ScheduledThreadPoolExecutor.java:294)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1142)

```

I hope that this information is useful!

I will try to look into this soon, I just don't have much spare time for
this project at the moment.

2017-12-11 22:27 GMT+01:00 jakobstenfeldt <notifications@github.com>:

> @kaikreuzer <https://github.com/kaikreuzer> I have tested the changes
> above and it looks OK, or at least not worse than before :) Only tested
> with Telldus Net.
>
> Bridge tellstick:telldus-live:2 "Tellstick Net" [refreshInterval=30000 ...]
>
> Some notes on the binding behaviour:
>
>    1. At start, the following error occurs. I guess its due to the fact
>    that the binding tries to load what's needed for running a Tellstick duo
>    (telldus-core) even though I have configured it to use Tellstick net.
>    (telldus-live)
>
> 2017-12-11 22:04:05.771 [ERROR] [t.i.d.TellstickBridgeDiscovery:78   ] - Could not load telldus core, please make sure Telldus is installed and correct 32/64 bit java. java.lang.UnsatisfiedLinkError: Unable to load library 'TelldusCore': Native library (win32-x86-64/TelldusCore.dll) not found in resource path (C:\Users\Jakob\openhab2-master\ws\.metadata\.plugins\org.eclipse.pde.core\.bundle_pool\plugins\org.eclipse.equinox.launcher_1.4.0.v20161219-1356.jar)
> 	at com.sun.jna.NativeLibrary.loadLibrary(NativeLibrary.java:271)
> 	at com.sun.jna.NativeLibrary.getInstance(NativeLibrary.java:398)
> 	at com.sun.jna.Library$Handler.<init>(Library.java:147)
> 	at com.sun.jna.Native.loadLibrary(Native.java:412)
> 	at com.sun.jna.Native.loadLibrary(Native.java:391)
> 	at org.tellstick.JNA$CLibrary.<clinit>(JNA.java:53)
> 	at org.tellstick.device.TellstickController.getControllers(TellstickController.java:56)
> 	at org.openhab.binding.tellstick.internal.discovery.TellstickBridgeDiscovery.listBridge(TellstickBridgeDiscovery.java:73)
> 	at org.openhab.binding.tellstick.internal.discovery.TellstickBridgeDiscovery.discoverController(TellstickBridgeDiscovery.java:66)
> 	at org.openhab.binding.tellstick.internal.discovery.TellstickBridgeDiscovery.startBackgroundDiscovery(TellstickBridgeDiscovery.java:55)
> 	at org.eclipse.smarthome.config.discovery.AbstractDiscoveryService.activate(AbstractDiscoveryService.java:385)
>
>
>    1. There is a warning shown multiple times. Two occurrences below. I
>    have 22 devices in my setup. Don't know why this happens...
>
> 2017-12-11 22:04:20.850 [WARN ] [.i.c.AbstractInvocationHandler:113  ] - Timeout of 5000ms exceeded while calling method 'ThingHandler.initialize()' on 'org.openhab.binding.tellstick.handler.TelldusDevicesHandler@229e749d'. Thread 'safeCall-1' (82) is in state 'WAITING'
> 	at sun.misc.Unsafe.park(Native Method)
> 	at java.util.concurrent.locks.LockSupport.park(LockSupport.java:175)
> 	at java.util.concurrent.locks.AbstractQueuedSynchronizer.parkAndCheckInterrupt(AbstractQueuedSynchronizer.java:836)
> 	at java.util.concurrent.locks.AbstractQueuedSynchronizer.acquireQueued(AbstractQueuedSynchronizer.java:870)
> 	at java.util.concurrent.locks.AbstractQueuedSynchronizer.acquire(AbstractQueuedSynchronizer.java:1199)
> 	at java.util.concurrent.locks.ReentrantLock$NonfairSync.lock(ReentrantLock.java:209)
> 	at java.util.concurrent.locks.ReentrantLock.lock(ReentrantLock.java:285)
> 	at org.eclipse.smarthome.core.thing.internal.ThingManager.thingUpdated(ThingManager.java:406)
> 	at org.eclipse.smarthome.core.thing.internal.ThingRegistryImpl.notifyTrackers(ThingRegistryImpl.java:222)
> 	at org.eclipse.smarthome.core.thing.internal.ThingRegistryImpl.notifyListenersAboutUpdatedElement(ThingRegistryImpl.java:145)
> 	at org.eclipse.smarthome.core.thing.internal.ThingRegistryImpl.notifyListenersAboutUpdatedElement(ThingRegistryImpl.java:1)
> 	at org.eclipse.smarthome.core.common.registry.AbstractRegistry.updated(AbstractRegistry.java:188)
> 	at org.eclipse.smarthome.core.common.registry.AbstractRegistry.updated(AbstractRegistry.java:1)
> 	at org.eclipse.smarthome.core.common.registry.AbstractProvider.notifyListeners(AbstractProvider.java:62)
> 	at org.eclipse.smarthome.core.common.registry.AbstractProvider.notifyListenersAboutUpdatedElement(AbstractProvider.java:86)
> 	at org.eclipse.smarthome.core.common.registry.AbstractManagedProvider.update(AbstractManagedProvider.java:132)
> 	at org.eclipse.smarthome.core.thing.internal.ThingManager$1$1.run(ThingManager.java:233)
> 	at org.eclipse.smarthome.core.thing.internal.ThingManager$1$1.run(ThingManager.java:1)
> 	at java.security.AccessController.doPrivileged(Native Method)
> 	...
> 2017-12-11 22:04:20.850 [WARN ] [.i.c.AbstractInvocationHandler:113  ] - Timeout of 5000ms exceeded while calling method 'ThingHandler.initialize()' on 'org.openhab.binding.tellstick.handler.TelldusDevicesHandler@d8d08b5'. Thread 'safeCall-3' (86) is in state 'WAITING'
> 	at sun.misc.Unsafe.park(Native Method)
> 	at java.util.concurrent.locks.LockSupport.park(LockSupport.java:175)
> 	at java.util.concurrent.locks.AbstractQueuedSynchronizer.parkAndCheckInterrupt(AbstractQueuedSynchronizer.java:836)
> 	at java.util.concurrent.locks.AbstractQueuedSynchronizer.acquireQueued(AbstractQueuedSynchronizer.java:870)
> 	at java.util.concurrent.locks.AbstractQueuedSynchronizer.acquire(AbstractQueuedSynchronizer.java:1199)
> 	at java.util.concurrent.locks.ReentrantLock$NonfairSync.lock(ReentrantLock.java:209)
> 	at java.util.concurrent.locks.ReentrantLock.lock(ReentrantLock.java:285)
> 	at org.eclipse.smarthome.core.thing.internal.ThingManager.thingUpdated(ThingManager.java:406)
> 	at org.eclipse.smarthome.core.thing.internal.ThingRegistryImpl.notifyTrackers(ThingRegistryImpl.java:222)
> 	at org.eclipse.smarthome.core.thing.internal.ThingRegistryImpl.notifyListenersAboutUpdatedElement(ThingRegistryImpl.java:145)
> 	at org.eclipse.smarthome.core.thing.internal.ThingRegistryImpl.notifyListenersAboutUpdatedElement(ThingRegistryImpl.java:1)
> 	at org.eclipse.smarthome.core.common.registry.AbstractRegistry.updated(AbstractRegistry.java:188)
> 	at org.eclipse.smarthome.core.common.registry.AbstractRegistry.updated(AbstractRegistry.java:1)
> 	at org.eclipse.smarthome.core.common.registry.AbstractProvider.notifyListeners(AbstractProvider.java:62)
> 	at org.eclipse.smarthome.core.common.registry.AbstractProvider.notifyListenersAboutUpdatedElement(AbstractProvider.java:86)
> 	at org.eclipse.smarthome.core.common.registry.AbstractManagedProvider.update(AbstractManagedProvider.java:132)
> 	at org.eclipse.smarthome.core.thing.internal.ThingManager$1$1.run(ThingManager.java:233)
> 	at org.eclipse.smarthome.core.thing.internal.ThingManager$1$1.run(ThingManager.java:1)
> 	at java.security.AccessController.doPrivileged(Native Method)
> 	...
> 2017-12-11 22:04:20.850 [WARN ] [.c.thing.internal.ThingManager:622  ] - Initializing handler for thing 'tellstick:switch:2:414128' takes more than 5000ms.
> 2017-12-11 22:04:20.851 [WARN ] [.c.thing.internal.ThingManager:622  ] - Initializing handler for thing 'tellstick:switch:2:1930308' takes more than 5000ms.
>
>
>    1. Got yet another warning...
>
> 2017-12-11 22:07:24.734 [WARN ] [.l.TelldusLiveDeviceController:285  ] - TimeoutException error in getjava.util.concurrent.TimeoutException: null
> 	at org.asynchttpclient.netty.NettyResponseFuture.get(NettyResponseFuture.java:169)
> 	at org.openhab.binding.tellstick.internal.live.TelldusLiveDeviceController.innerCallRest(TelldusLiveDeviceController.java:308)
> 	at org.openhab.binding.tellstick.internal.live.TelldusLiveDeviceController.callRestMethod(TelldusLiveDeviceController.java:282)
> 	at org.openhab.binding.tellstick.internal.live.TelldusLiveBridgeHandler.updateDevices(TelldusLiveBridgeHandler.java:132)
> 	at org.openhab.binding.tellstick.internal.live.TelldusLiveBridgeHandler.refreshDeviceList(TelldusLiveBridgeHandler.java:119)
> 	at org.openhab.binding.tellstick.internal.live.TelldusLiveBridgeHandler$1.run(TelldusLiveBridgeHandler.java:70)
> 	at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:511)
> 	at java.util.concurrent.FutureTask.runAndReset(FutureTask.java:308)
> 	at java.util.concurrent.ScheduledThreadPoolExecutor$ScheduledFutureTask.access$301(ScheduledThreadPoolExecutor.java:180)
> 	at java.util.concurrent.ScheduledThreadPoolExecutor$ScheduledFutureTask.run(ScheduledThreadPoolExecutor.java:294)
> 	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1142)
>
>
> I hope that this information is useful!
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/openhab/openhab2-addons/issues/2927#issuecomment-350863545>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/AAZhJmz0hf5p9Eno1Hi3aBcoDeDm2qlBks5s_Z46gaJpZM4Q5DfQ>
> .
>

--- a/addons/binding/org.openhab.binding.tellstick/src/main/java/org/openhab/binding/tellstick/internal/core/TelldusCoreBridgeHandler.java
+++ b/addons/binding/org.openhab.binding.tellstick/src/main/java/org/openhab/binding/tellstick/internal/core/TelldusCoreBridgeHandler.java
@@ -78,8 +78,14 @@ public class TelldusCoreBridgeHandler extends BaseBridgeHandler
     @Override
     public void dispose() {
         logger.debug("Telldus Core Handler disposed.");
-        deviceController.dispose();
-        eventHandler.remove();
+        if (deviceController != null) {
+            deviceController.dispose();
+            deviceController = null;
+        }
+        if (eventHandler != null) {
+            eventHandler.remove();
+            eventHandler = null;
+        }
         clearDeviceList();
         initialized = false;
         JNA.CLibrary.INSTANCE.tdClose();
@@ -108,10 +114,14 @@ public class TelldusCoreBridgeHandler extends BaseBridgeHandler
         TellstickBridgeConfiguration configuration = getConfigAs(TellstickBridgeConfiguration.class);
         init(configuration.libraryPath);
 
-        rescanTelldusDevices();
-        setupListeners();
-        setupDeviceController(configuration);
-        updateStatus(ThingStatus.ONLINE);
+        scheduler.submit(() -> {
+            rescanTelldusDevices();
+            setupListeners();
+            setupDeviceController(configuration);
+            updateStatus(ThingStatus.ONLINE);
+        });
+
+        updateStatus(ThingStatus.UNKNOWN);
     }
 
     private void setupDeviceController(TellstickBridgeConfiguration configuration) {
@@ -191,6 +201,8 @@ public class TelldusCoreBridgeHandler extends BaseBridgeHandler
     }
 
     public void clearDeviceList() {
+        deviceList.clear();
+        sensorList.clear();
     }
 
     private Device getDevice(String id, List<TellstickDevice> devices) {

